### PR TITLE
Feat/v2 geostore routes

### PR DESCRIPTION
**Pull request by 01painadam**
# Prep for AOI

Adds /v2 endpoints that allow for:

- new schema required by gfw AOI entities (`tags`, `status`, `public`, and more in the future)
- new endpoints for handling updates to area entities
- endpoint for bulk updates of several AOIs given a list of `geostore_id`'s

To update by `geostore` id (NOTE: not the entities `id`) use a payload like:

``` 
{
    'geostores': [<geostore_1>, geostore_2, ..., <geostore_n>],
    'update_params': 
    {
        <update key/values>
    }
}

```

### Known issues

Before merging we need to investigate how the 'Teams' structure works. For example, when deleting an area entity the service checks the Team that the area's owner belongs to, and removes the areas association to that team (before deleting).

This may cause issues down the line if FW users have their shapes in AOI.

One solutions is just to remove the fail state from the delete route. If the area doesnt belong to a team, just delete it (instead of throwing an error).

### CODE REVIEW

File: app/src/routes/api/v2/area.router.js [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+    static async updateByGeostore(ctx) {
+        const geostores = ctx.request.body.geostores || [];
+        const update_params = ctx.request.body.update_params || {};
+        logger.info('Updating geostores: ', geostores);
+        logger.info('Updating with params: ', update_params);
+
+        // validate update json
+        update_params.updatedDate = Date.now;
+
+        const area = await AreaModel.updateMany(
```
Comment (edbrett) [inline app/src/routes/api/v2/area.router.js]: Didn't know about this method :)

File: app/src/routes/api/v2/area.router.js [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+        logger.info(`Deleting area with id ${ctx.params.id}`);
+        const userId = ctx.state.loggedUser.id;
+        let team = null;
+        try {
+            team = await TeamService.getTeamByUserId(userId);
+        } catch (e) {
+            logger.error(e);
+            ctx.throw(500, 'Team retrieval failed.');
+        }
+        if (team && team.areas.includes(ctx.params.id)) {
```
Comment (edbrett) [inline app/src/routes/api/v2/area.router.js]: DELETE is still dependant on teams. Likely we want a separate delete for FW?

File: app/src/routes/api/v2/area.router.js [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+        return;
+    }
+    if (area.userId !== ctx.state.loggedUser.id && area.userId !== ctx.request.body.userId) {
+        ctx.throw(403, 'Not authorized');
+        return;
+    }
+    await next();
+}
+
+router.get('/', loggedUserToState, AreaRouterV2.getAll);
```
Comment (edbrett) [inline app/src/routes/api/v2/area.router.js]: Do we not want to check permissions on GET all?

File: app/src/validators/area.validatorV2.js [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    static notEmptyString(property) {
+        if (typeof property === 'string' && property.length > 0) {
+            return true;
+        }
+        return false;
+    }
+
+    static isArray(property) {
+        if (property instanceof Array) {
+            // const invalid = property.filter(str =>
```
Comment (edbrett) [inline app/src/validators/area.validatorV2.js]: Ugly. Legacy?

File: dev.env.sample [hunk 5]
```diff
@@ -0,0 +1,6 @@
+CT_URL=http://mymachine:9000
+LOCAL_URL=http://mymachine:4200
+CT_TOKEN=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6Im1pY3Jvc2VydmljZSIsImNyZWF0ZWRBdCI6IjIwMTYtMDktMTQifQ.IRCIRm1nfIQTfda_Wb6Pg-341zhV8soAgzw7dd5HxxQ
```
Comment (edbrett) [inline dev.env.sample]: Is this ok?

File: app/src/routes/api/v2/area.router.js (hunk 2 above)
Comment (01painadam) [reply app/src/routes/api/v2/area.router.js]: yep, we should.

File: app/src/routes/api/v2/area.router.js (hunk 3 above)
Comment (01painadam) [reply app/src/routes/api/v2/area.router.js]: Depends on privacy and whether there is a use-case in the front-end for fetch all areas.

This GET method is more like "get all of _my_ AOIs" for a particular user. I think if you have an admin OATH then that _my_ areas is everything. Will have to test in staging.

File: app/src/validators/area.validatorV2.js (hunk 4 above)
Comment (01painadam) [reply app/src/validators/area.validatorV2.js]: Working on it

File: dev.env.sample (hunk 5 above)
Comment (01painadam) [reply dev.env.sample]: Nope, added git ignore

**Pull request merged**